File: benchmarks/pow-1K/main.go
package main

import (
	"fmt"
	"math"
)
import "time"

func main() {
	var before = time.Now().UnixNano()

	for i := 0; i < 1000; i++ {
		var y = 1.1 + (float64(i) / 1000.0)
		math.Pow(2, y)
	}

	var after = time.Now().UnixNano()

	fmt.Println(after - before)
}

File: benchmarks/pow-1K/main.zig
const std = @import("std");

pub fn main() !void {
    const before: i128 = std.time.nanoTimestamp();

    for (0..999) |i| {
        const y = 1.1 + (@as(f32, @floatFromInt(i)) / 1000.0);
        _ = std.math.pow(f64, 2.0, y);
    }

    const after: i128 = std.time.nanoTimestamp();

    std.debug.print("{}", .{after - before});
}